DESIGNME.md

```diff
@@ -0,0 +1,119 @@
+
+**Step Two**
+
+1.  Everybody loves to copy paste
+    When you start with d3, you get a long-ass list of scatter plots and you copy
+    paste one and pray it works.
+2.  Functionality is locked up in bl.ocks and gists -- wrong. Gists aren't the
+    right format for code of this kind.
+3.  With d3-as-it-has-been, functionality is not componentized. 
+4.  So we'll make: components as discrete, manageable, configurable pieces
+5.  There are a lot of other charting libraries out there; this one is
+    different, especially if you're doing this from the beginning
+6.  Sensible defaults
+
+
+### Design / Welcome to the Neighborhood Questions
+
+**Preface**: This is not my first rodeo; it's not even really my first
+  JavaScript Rodeo, really. But: I'm a server-dweller. I work a lot in compiled
+  languages, with a heavy dose of Python. Some of these thoughts and questions
+  might be pretty far on the n00b end of the spectrum. Learn me a thing. I can
+  keep up.
+
+
+#### Question One: Composition
+
+How granular do we hope to get with the composed nature of `react`? I've worked
+with `react` some; I use composition-the-programming-technique a freaking
+lot. When somebody says, "let's use `react` to replace `d3`'s hideous DOM
+model", what I think of is a heirarchy a little like:
+
+```
+module: chart
+constituent modules:
+- line
+- grid (made of line again)
+- point (made of a shape and an optional label)
+- legend
+```
+
+But now we're getting in to the business of using `react` to replace `d3`'s
+logic, instead of just it's math. Another, maybe/probably better way to do this
+would look more like going down Michael Bostock's npm profile page and making
+one `victory`  module for every `d3` module. Chart, then, would just be:
+
+```
+module: chart
+```
+
+I'm kinda hoping we're planning on the second one. Y'all?
+
+This also feeds in to the next question.
+
+#### Question Two: The Thousand Modules Problem
+
+This has been on my mind a lot since coming aboard to pitch in with
+`victory`. Let's see if I can articulate this well, but briefly.
+
+In server-side languages land, I might compile a Java jar and it might be 43
+megabytes and I might laugh at how small that is. The compiled code that runs my
+web site is something over two hundred meg. Chatting with Colin the other night,
+he said very emphatically, "you know, most people are pulling in all of d3 just
+to use one little thing and it's saddling them with like, 150k of
+dependencies. 150k! Of stuff they aren't using!"
+
+So that was eye-opening. The world of terabyte-scale data transformation is a
+little different than the world of browsers.
+
+But, now we have two different things to consider: maintainability verses, for
+lack of a better phrase, small modules. We could make a single `victory` repo
+containing every `victory` component. You'd use the whole blob, and it'd be `d3`
+wearing a heavy coat. We could make a one-to-one correspondence, `d3` module ->
```

Comment:
*nods* I'm getting my head around this, and the mapping between repos and modules -- which may or may not be 1:1